spring-data-jdbc/src/main/java/org/springframework/data/jdbc/repository/support/JdbcRepositoryFactoryBean.java

```diff
@@ -157,9 +158,14 @@ public void setBeanFactory(BeanFactory beanFactory) {
 	@Override
 	public void afterPropertiesSet() {
 
-		Assert.state(this.mappingContext != null, "MappingContext is required and must not be null");
 		Assert.state(this.converter != null, "RelationalConverter is required and must not be null");
 
+		if (mappingContext == null) {
```

Comment:
We can actually get the MappingContext from `JdbcConverter` and therefore, we don't require any further lookup.

I suggest that you also have a look at `setQueryMappingConfiguration(…)` and remove `@Autowired(required = false)` from there and fall back to a lookup (via `ObjectProvider`) in `afterPropertiesSet`.